File: js/src/shi_mart.js
class Item {
  constructor(name, sellIn, quality) {
    this.name = name;
    this.sellIn = sellIn;
    this.quality = quality;
  }
}

class Shop {
  constructor(items=[]){
    this.items = items;
  }

  // 1. If canned beans, nothing
  // 2. Everything else, --
  updateQuality() {
    for (let i = 0; i < this.items.length; i++) {
      // switch
      // {
        // if(this.items[i].name == 'Canned Beans')
        //    coninue;
        //  this.items[i].sellIn = this.items[i].sellIn - 1;
        // if(....){
          // }
          // 
          // }
          // }

      if (this.items[i].name != 'Canned Beans') {
        this.items[i].sellIn = this.items[i].sellIn - 1;
      }

      // Aged Brie Alone
      if (this.items[i].name == 'Aged Brie') {
        if (this.items[i].quality < 50) {
          this.items[i].quality = this.items[i].quality + 1;
        }
      } else {
        if (this.items[i].quality > 0 && this.items[i].name != 'Canned Beans' && this.items[i].name != 'Baked Sourdough Bread') {
          this.items[i].quality = this.items[i].quality - 1;
        }
        else if(this.items[i].name == 'Baked Sourdough Bread')
        {
          if(this.items[i].quality > 1)
            this.items[i].quality = this.items[i].quality - 2;
          else if(this.items[i].quality == 1)
            this.items[i].quality = this.items[i].quality - 1;
        } 
      }

      if (this.items[i].sellIn < 0) {
        if (this.items[i].name == 'Aged Brie') {
          if (this.items[i].quality < 50) {
            this.items[i].quality = this.items[i].quality + 1;
          }
        } 
        
        else {
          if (this.items[i].quality > 0 && this.items[i].name != 'Canned Beans') {
            if(this.items[i].name != 'Baked Sourdough Bread')
              this.items[i].quality = this.items[i].quality - 1;
            else if(this.items[i].name == 'Baked Sourdough Bread')
            {
              if(this.items[i].quality > 1)
                this.items[i].quality = this.items[i].quality - 2;
              else if(this.items[i].quality == 1)
                this.items[i].quality = this.items[i].quality - 1;
            } 
          }
        }
      }

    }
    return this.items;
  }
}

module.exports = {
  Item,
  Shop
}

File: python/shi_mart.py
# -*- coding: utf-8 -*-

class ShiMart(object):

    def __init__(self, items):
        self.items = items

    def update_quality(self):
        for item in self.items:
            if item.name != "Aged Brie":
                if item.quality > 0:
                    if item.name != "Canned Beans":
                        item.quality = item.quality - 1
            else:
                if item.quality < 50:
                    item.quality = item.quality + 1
            if item.name != "Canned Beans":
                item.sell_in = item.sell_in - 1
            if item.sell_in < 0:
                if item.name != "Aged Brie":
                    if item.quality > 0:
                        if item.name != "Canned Beans":
                            item.quality = item.quality - 1
                else:
                    if item.quality < 50:
                        item.quality = item.quality + 1


class Item:
    def __init__(self, name, sell_in, quality):
        self.name = name
        self.sell_in = sell_in
        self.quality = quality

    def __repr__(self):
        return "%s, %s, %s" % (self.name, self.sell_in, self.quality)
